src/test/groovy/acceptance/steps/NameMatchingSteps.groovy

```diff
@@ -0,0 +1,327 @@
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options
+import static com.jayway.restassured.RestAssured.given
+import static java.time.format.DateTimeFormatter.ISO_DATE
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE_TIME
+import static org.junit.Assert.*
+
+@SuppressFBWarnings([
+        "SE_NO_SERIALVERSIONID",
+        "ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD"])
+@ContextConfiguration
+@SpringBootTest(classes = ServiceRunner.class,
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+        properties = [
+                "pttg.audit.url=http://localhost:1111",
+                "base.hmrc.url=http://localhost:2222",
+                "base.hmrc.access.code.url=http://localhost:3333"
+        ])
+class NameMatchingSteps implements ApplicationContextAware {
+    private static final RESPONSE_SESSION_KEY = "IndividualMatchingResponse"
+    private static final IGNORE_JSON_ARRAY_ORDER = true
+    private static final IGNORE_EXTRA_ELEMENTS = true
+    private static final INDIVIDUAL_MATCHING_ENDPOINT = "/individuals/matching/"
+    private static final MATCH_ID = "s87654321"
+
+    private static final AUDIT_MOCK_SERVICE = new WireMockServer(options().port(1111))
+    private static final HMRC_MOCK_SERVICE = new WireMockServer(options().port(2222))
+    private static final ACCESS_KEY_MOCK_SERVICE = new WireMockServer(options().port(3333))
+
+    private static isSetup = false
+
+    @LocalServerPort
+    private int port
+
+    @Before
+    void before() throws Exception {
+        if (!isSetup) {
+            HMRC_MOCK_SERVICE.start()
+
+            AUDIT_MOCK_SERVICE.start()
+            ACCESS_KEY_MOCK_SERVICE.start()
+
+            setupAuditStub()
+            setupAccessCodeStub()
+
+            RestAssured.port = port
+            isSetup = true
+        }
+
+        setupStubsForOtherHmrcCalls()
+    }
+
+    private static StubMapping setupAuditStub() {
+        AUDIT_MOCK_SERVICE
+                .stubFor(post(urlMatching("/audit.*"))
+                .willReturn(aResponse().withStatus(200)))
+    }
+
+    private static void setupAccessCodeStub() {
+        def accessCodeJsonResponse = JsonOutput.toJson([
+                code  : 'TestCode',
+                expiry: ISO_LOCAL_DATE_TIME.format(LocalDateTime.now().plusHours(5))
+        ])
+        ACCESS_KEY_MOCK_SERVICE.stubFor(get(urlMatching("/access"))
+                .willReturn(aResponse()
+                .withHeader("Content-Type", "application/json")
+                .withBody(accessCodeJsonResponse)
+                .withStatus(200)))
+    }
+
+    @Override
+    void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+        DispatcherServlet dispatcherServlet = applicationContext.getBean("dispatcherServlet") as DispatcherServlet
+        dispatcherServlet.setThrowExceptionIfNoHandlerFound(true)
```

Comment:
It feels wrong to be asking the DispatcherServlet to behave differently in the test to the real application - however there may be a good reason for this.  Please can you do a little digging to make sure this is justified?

Reply:
Pulled this through without thorough checking - was commented `required for @ControllerAdvice to work` but this doesn't seem to be the case. I have removed.